lib/frame.rb

```diff
@@ -0,0 +1,35 @@
+class Frame
+    attr_reader :contents
+  
+    def initialize
+      @contents = []
+    end
+  
+    def raw_score
+      @contents.sum
+    end
+  
+    def record_roll(roll)
+      @contents << roll
+    end
+  
+    def rolls
+      @contents.length
+    end
+  
+    def spare?
+      if (@contents.sum == 10) && (@contents.length == 2)
+        true
+      else
+        false
+      end
+    end
+  
+    def strike?
+      if (@contents.sum == 10) && (@contents.length == 1)
+        true
+      else
+        false
+      end
+    end
+  end
```

Comment:
Really nice and clean!! For the predicate methods spare? and strike?, I think you can just make the statement, and the method will return true or false without it needing to be in an if statement, eg. 
def spare?
  (@contents.sum == 10) && (@contents.length == 2)
end
That should return true if it's true and false if not, without needing the if... else.